src/main/kotlin/gg/essential/vigilance/Vigilant.kt

```diff
@@ -142,10 +142,38 @@ abstract class Vigilant @JvmOverloads constructor(
             error("Dependency must be a boolean PropertyType!")
         }
 
-        property.dependsOn = dependency
+        property.dependsOn = { dependency.getAsBoolean() }
         dependency.hasDependants = true
     }
 
+    /**
+     * If the dependency is based on another property, make sure to also call setAsDependency
+     */
+    fun addCustomDependency(dependingPropertyName: String, stateProvider: () -> Boolean) {
+        propertyCollector.getProperty(dependingPropertyName)!!.dependsOn = stateProvider
+    }
+
+    /**
+     * If the dependency is based on another property, make sure to also call setAsDependency
+     */
+    fun addCustomDependency(dependingField: Field, stateProvider: () -> Boolean) {
+        propertyCollector.getProperty(dependingField)!!.dependsOn = stateProvider
+    }
+
+    /**
+     * Should be used when a property has a custom dependency based on this property
+     */
+    fun setAsDependency(dependencyPropertyName: String) {
+        propertyCollector.getProperty(dependencyPropertyName)!!.hasDependants = true
+    }
+
+    /**
+     * Should be used when a property has a custom dependency based on this property
+     */
+    fun setAsDependency(dependencyField: Field) {
+        propertyCollector.getProperty(dependencyField)!!.hasDependants = true
+    }
```

Comment:
What's the point of having separate methods for creating the dependency and then stating which field/property it depends on?

Reply:
When you add a custom dependency, it doesn't need to be based on a property. For example, you could make a property be displayed only when you're connected to a certain server.
`setAsDependency` tells Vigilance to re-evaluate the visibility when the dependency property has changed. If `setAsDependency` hasn't been called, Vigilance will assume that the dependency doesn't care about any other properties.

Now that I think more about it, it would probably also be possible for `addCustomDependency` functions to have the dependency field as an optional argument.